core/tee/tee_svc_cryp.c

```diff
@@ -3420,7 +3422,12 @@ TEE_Result tee_svc_asymm_verify(uint32_t state,
 	if (res != TEE_SUCCESS)
 		goto out;
 
-	if (data_len != hash_size) {
+    	/*
+    	 * Depending on the DSA algorithm, the size of the hash to consider
+	 * could be less or equal than he whole size of the
+	 * hash algorithm involved (related to the subprime).
```

Comment:
Yes. For DSA-SH256, the hash value generated is coded on 256 bits, but the DSA  (encrypted hash value) could be either on 224bits or 256 bits (depending on the subprime).